docs/DeveloperGuide.md

```diff
@@ -6,33 +6,230 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### The Overall UML Diagram:
 
+![UML Diagram](assets/images/mathGenius.png)    
+
```

Comment:
Could the symbols you used to show visibility of attributes and methods be wrong perhaps? 
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/108778858/033dda2f-e28d-4d63-a650-34578c45efce)

Refer to the example given here:
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/108778858/bd557ef1-06b1-4400-855b-0a9dffe14664)